src/main/java/com/google/sps/data/DatastoreAccess.java

```diff
@@ -147,10 +147,15 @@ public boolean saveUser(UserAccount user) {
     Query query =
         new Query(ParameterConstants.ENTITY_TYPE_USER_ACCOUNT)
             .setFilter(
-                new Query.FilterPredicate(
-                    ParameterConstants.USER_TYPE,
-                    Query.FilterOperator.EQUAL,
-                    UserType.MENTOR.ordinal()));
+                Query.CompositeFilterOperator.and(
+                    new Query.FilterPredicate(
+                        ParameterConstants.USER_TYPE,
+                        Query.FilterOperator.EQUAL,
+                        UserType.MENTOR.ordinal()),
+                    new Query.FilterPredicate(
+                        ParameterConstants.MENTOR_TYPE,
+                        Query.FilterOperator.EQUAL,
+                        mentee.getDesiredMentorType().ordinal())));
```

Comment:
where are the bounds (we wanted to say return no more than 100 results but no less than ... results)

Reply:
the bound is done a few lines below